### ISSUE TITLE
TinyBert : Why applying temperature to student only ?

### ISSUE THREAD

**Issue by Colanim** [huawei-noah/Pretrained-Language-Model#11]
From the [paper ](https://arxiv.org/pdf/1909.10351.pdf) :

![image](https://user-images.githubusercontent.com/43774355/70582784-f9835b80-1bfe-11ea-9431-16a60fb25584.png)

Temperature is not applied to teacher's logits.

---

But in the code :

https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/task_distill.py#L967-L968

Temperature is applied to both student and teacher's logits.

---

**Should temperature be applied to both student and teacher's logits ?**

**Comment by nlpBeginner** [comment]
> From the [paper ](https://arxiv.org/pdf/1909.10351.pdf) :
> 
> ![image](https://user-images.githubusercontent.com/43774355/70582784-f9835b80-1bfe-11ea-9431-16a60fb25584.png)
> 
> Temperature is not applied to teacher's logits.
> 
> But in the code :
> 
> https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/task_distill.py#L967-L968
> 
> Temperature is applied to both student and teacher's logits.
> 
> **Should temperature be applied to both student and teacher's logits ?**

We use the temperature=1 in our paper, thus the different formulations are same for TinyBERT training.  The formulation in the code is correct for prediction-layer distillation and we will fix the typo in our paper. Thanks!
